metricflow-semantics/metricflow_semantics/model/semantics/linkable_spec_resolver.py

```diff
@@ -188,6 +188,10 @@ def __init__(
 
         logger.debug(LazyFormat(lambda: f"Building valid group-by-item indexes took: {time.time() - start_time:.2f}s"))
 
+    def _semantic_model_is_scd(self, semantic_model: SemanticModel) -> bool:
```

Comment:
Might wanna cache this in a dict to avoid iterating dimensions over and over and over again.